### ISSUE TITLE
Dark Matter JFactory Map 

### ISSUE THREAD

**Issue by katrinstreil** [gammapy/gammapy#2769]
The calculation of the differential J-Factor is by defintion over the line of sight ds and not over the radius dr. When integrating over dr, the integral is missing the substitution of ds:
math::
            \frac{d J}{d \Omega} =
            \int_{LoS}  ds \rho(r) = 
            \int_{\r_min}^{\r_max} dr \frac{d s}{d r} \rho(r) =
            \int_{\r_min}^{\r_max} dr frac{r}{sqrt{r**2- d**2* sin(\theta)**2}} \rho(r)

since r = sqrt{d**2 + s**2 -2dl cos(\theta)}, where d is the distance to the GC and \theta the separation angle.
I also found the integral limits to be incorrect. Integrating from rmin = separation.rad * self.distance to  rmax = self.distance accounts only for the line of sight to the object at distance. But the line of sight continues beyond it. 

In my solution I included the derivative \frac{d s}{d r} in the  _eval_squared method of the corresponding profile class. Therefore the parameter separation has to be passed to the method integral in the corresponding profile class, since it calls  _eval_squared.

If there is any need, I can upload my script.

**Comment by Bultako** [comment]
@kstre 

Thanks for spotting this !

I think that the calculation of J-Factor maps in Gammapy was introduced more as a coding exercise than as a feature to compute accurate values. For this, we know that people from the dark matter community very often uses [CLUMPY](https://clumpy.gitlab.io/CLUMPY), which provides precise values with rigorous calculations and multiple fine-tuning parameters.

Having said that, you're right that the [definition of the differential J-factor](https://docs.gammapy.org/0.16/api/gammapy.astro.darkmatter.JFactory.html#gammapy.astro.darkmatter.JFactory.compute_differential_jfactor)  is not correct (at least in the documentation, and this has to be fixed) and that calculation of J-factor maps could be actually done in a more proper way, even if this is not the ultimate goal of Gammapy. Could you please paste a **gist link** to your script or the piece of code that shows how you fix it, please? So we can continue talking on this and agree on how to proceed.

**Comment by katrinstreil** [comment]
Hi, sorry for the (very) slow response!
I created a gist (https://gist.github.com/katrinstreil/eeee9f2969dbc037888f876d5f4e5b9d) with my solution in it. I modified gammapy.astro.darkmatter.profiles.py and gammapy.astro.darkmatter.utils.py to correct for the integral limits and the substitution part. In addition, there is a notebook, which computes and compares the J-Factor map with the NFW profile computed with the original and the modified code.

I would be very happy to discuss this further! 
 [https://gist.github.com/katrinstreil/eeee9f2969dbc037888f876d5f4e5b9d](url)
